### PR TITLE
Enhance Go plugin and tasks for cross-platform compatibility

### LINKED ISSUE #12721: Investigate and resolve issues running rosetta tests on Windows
### Description

There have been issues while we ran tests on windows machine. We need to investigate the issue and resolve any problems.

### Steps to reproduce

run `./gradlew rosetta:test`
identify any errors

### Additional context

_No response_

### Hedera network

_No response_

### Version

v.0.147.0

### Operating system

Windows

### PR DESCRIPTION
**Pull request by karthik120710**
Description:

Add cross-platform support for the Go plugin on Windows.
* Updated binary naming in go-conventions.gradle.kts to support Windows executables
* Modified GoPlugin.kt to set the correct Go binary path based on the operating system
* Adjusted GoSetup.kt to download the appropriate Go distribution format (zip for Windows, tar.gz for others) and added zip extraction support
* Refactored Docker mount path handling in db.go to ensure compatibility with Windows paths
* Added new tests in OpcodeActionTracerTest.java to verify storage behavior across multiple opcode executions

Java test : 
  Tests (web3 module)                                                                                                                                
  - Added shouldReturnUpdatedSlotValueWhenCountIsUnchangedBetweenOpcodes — verifies that when the same storage slot is overwritten in-place (access    
  count stays the same between opcodes), the tracer captures the updated value rather than the stale one                                           
  - Added shouldTrackStorageThroughMultipleSequentialOverwritesOfSameSlot — verifies correct slot value tracking across three consecutive overwrites of
   the same slot within a single transaction                                                                                                           
                                                                                                                                                       
  These tests cover the edge case where StorageAccessTracker.put replaces an existing entry (keeping count at 1), ensuring the opcode tracer always  
  reflects the latest slot value per opcode. 


Related issue(s):

Fixes #12721

Checklist
- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: rosetta/test/db/db.go [hunk 1]
```diff
@@ -109,7 +108,7 @@ func (d dbParams) toConfig() config.Db {
 
 // CleanupDb cleans the data written to the db during tests
 func CleanupDb(db *sql.DB) {
-	filename := filepath.Clean(path.Join(moduleRoot, dbCleanupScript))
+	filename := filepath.Join(moduleRoot, dbCleanupScript)
 	script, err := os.ReadFile(filename)
```
Comment (martingeorgiev1) [inline rosetta/test/db/db.go]: this is causing the gosec CI failure (G304 / CWE-22)  

  filename is flagged as a variable path, but it's built from `runtime.Caller(0) `+ a hardcoded constant,not user input, so it's a false positive. Fix by adding a nosec suppression on line 112: 
`script, err := os.ReadFile(filename) // #nosec G304  `

File: web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java [hunk 2]
```diff
@@ -815,6 +815,79 @@ private MessageFrame.Builder messageFrameBuilder(final Address recipientAddress,
...
+        frame = setupInitialFrame(opcodeContext);
+
+        final var slotKey = UInt256.ZERO;
+        final var valueV1 = UInt256.ONE;
+        final var valueV2 = UInt256.valueOf(42L);
+
+        // Opcode X: slot K1 written to V1 (access count = 1)
+        final var usage1 = new TxStorageUsage(
+                List.of(new StorageAccesses(
+                        ContractID.DEFAULT, List.of(new StorageAccess(slotKey, UInt256.ZERO, valueV1)))),
```
Comment (martingeorgiev1) [inline web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java]: `UInt256.ZERO` appears as both the slot key and the previous value — ambiguous. Name the second one:
```
final var previousValue = UInt256.ZERO;
new StorageAccess(slotKey, previousValue, valueV1) 
```
Same for lines 869, 873, 877

File: web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java [hunk 3]
```diff
@@ -815,6 +815,79 @@ private MessageFrame.Builder messageFrameBuilder(final Address recipientAddress,
                 .contextVariables(Map.of(ContractCallContext.CONTEXT_NAME, contractCallContext));
     }
 
+    @Test
```
Comment (martingeorgiev1) [inline web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java:825]: These tests cover a storage-tracking bug in OpcodeActionTracer, not Windows compatibility. Can you clarify the connection to this PR? If unrelated, they should be in a separate PR tied to the fix or add the reason to the PR description.

File: buildSrc/src/main/kotlin/go-conventions.gradle.kts [hunk 4]
```diff
@@ -14,9 +14,14 @@ apply<GoPlugin>()
 
 val go = project.extensions.getByName<GoExtension>("go")
 
+val isWindows = System.getProperty("os.name").lowercase().contains("windows")
```
Comment (martingeorgiev1) [inline buildSrc/src/main/kotlin/go-conventions.gradle.kts]: go.os is already set by detectOs() in GoPlugin.kt — no need to re-detect via system property here. Use the extension instead:                                             
  val isWindows = go.os == "windows"

File: buildSrc/src/main/kotlin/plugin/go/GoSetup.kt [hunk 5]
```diff
@@ -36,7 +40,29 @@ open class GoSetup : DefaultTask() {
...
+            }
+        }
+    }
+
+    fun decompressZip(source: File, destDir: File) {
+        ZipInputStream(FileInputStream(source)).use { zip ->
+            destDir.mkdirs()
+            var entry = zip.nextEntry
+            while (entry != null) {
+                val file = destDir.resolve(entry.name)
```
Comment (martingeorgiev1) [inline buildSrc/src/main/kotlin/plugin/go/GoSetup.kt:57]: Zip-slip vulnerability. A zip entry with a path like ../../evil can write files outside the destination directory. Add a canonical path check before extracting:
```
  val file = destDir.resolve(entry.name)                                                                                                                                    
  require(file.canonicalPath.startsWith(destDir.canonicalPath + File.separator)) {
      "Zip entry escapes destination directory: ${entry.name}"                                                                                                              
  }   
```

File: web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java [hunk 6]
```diff
@@ -815,6 +815,79 @@ private MessageFrame.Builder messageFrameBuilder(final Address recipientAddress,
...
+    @DisplayName("should return updated slot value when access count is unchanged between opcodes")
+    void shouldReturnUpdatedSlotValueWhenCountIsUnchangedBetweenOpcodes() {
+        // Given - storage enabled
```
Comment (martingeorgiev1) [inline web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java:828]: In both `shouldReturnUpdatedSlotValueWhenCountIsUnchangedBetweenOpcodes` and `shouldTrackStorageThroughMultipleSequentialOverwritesOfSameSlot ` 
The `TxStorageUsage` construction is copy-pasted identically across both tests. Extract a private helper to remove the duplication:                                         
```
  private TxStorageUsage makeUsage(UInt256 slotKey, UInt256 value) {                                                                                                        
      return new TxStorageUsage(                                    
          List.of(new StorageAccesses(                                                                                                                                      
              ContractID.DEFAULT, List.of(new StorageAccess(slotKey, UInt256.ZERO, value)))),
          Set.of());                                                                                                                                                        
  } 
```

**Review by martingeorgiev1 — CHANGES_REQUESTED** [review]
Code is successfully passing the tests on **windows** but needs some changes

File: web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java (hunk 3 above)
Comment (karthik120710) [reply web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java:825]: @martingeorgiev1  can i revert java related changes in these pr and create a separate pr for the java related changes .

**Comment by karthik120710** [comment]
@martingeorgiev1 done the changes that u have suggested .

**Comment by karthik120710** [comment]
@martingeorgiev1 @nirbosl Hi, could you please trigger the CI checks for this PR?

File: buildSrc/src/main/kotlin/plugin/go/GoSetup.kt [hunk 7]
```diff
@@ -36,17 +40,47 @@ open class GoSetup : DefaultTask() {
...
+    /** Extracts a tar.gz archive into the given destination directory. */
     fun decompressTgz(source: File, destDir: File) {
         TarArchiveInputStream(GZIPInputStream(FileInputStream(source)), "UTF-8").use {
             destDir.mkdirs()
             var entry = it.nextEntry
 
             while (entry != null) {
                 val file = destDir.resolve(entry.name)
+                require(file.canonicalPath.startsWith(destDir.canonicalPath + File.separator)) {
+                    "Zip entry escapes destination directory: ${entry.name}"
```
Comment (martingeorgiev1) [inline buildSrc/src/main/kotlin/plugin/go/GoSetup.kt]: NIT: The error message says "Zip entry escapes destination directory" but this is inside decompressTgz, not the zip extractor. Should be "Tar entry escapes destination directory..".

File: web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java (hunk 3 above)
Comment (martingeorgiev1) [reply web3/src/test/java/org/hiero/mirror/web3/evm/contracts/execution/traceability/OpcodeActionTracerTest.java:825]: You can keep this as is, just update the PR description.

**Review by martingeorgiev1 — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/kotlin/plugin/go/GoSetup.kt (hunk 7 above)
Comment (karthik120710) [reply buildSrc/src/main/kotlin/plugin/go/GoSetup.kt]: @martingeorgiev1 i have changed zip entry to Tar entry and added test web3 module in pr  discription .

**Review by martingeorgiev1 — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM. Thanks for your contribution!

**Comment by steven-sheehy** [comment]
Just noticed that commits are not GPG signed, which is a requirement. Can you check and fix?

**Comment by karthik120710** [comment]
@steven-sheehy signed my commits .
